d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -1044,33 +1108,32 @@ final class RpcRetryTask implements Runnable
     public void run()
     {
       startRpcStreamLocal();
-      for (ResourceType originalType : getResourceSubscribers().keySet())
+
+      for (ResourceType originalType : ResourceType.values())
       {
         Set<String> resources = new HashSet<>(getResourceSubscriberMap(originalType).keySet());
-        if (resources.isEmpty())
-        {
-          continue;
-        }
+        boolean isGlobCollection = shouldSubscribeUriGlobCollection(originalType);
+        ResourceType adjustedType = isGlobCollection ? ResourceType.D2_URI : originalType;
 
-        ResourceType adjustedType;
-        if (shouldSubscribeUriGlobCollection(originalType))
+        if (isGlobCollection)
         {
           resources = resources.stream()
               .map(GlobCollectionUtils::globCollectionUrlForClusterResource)
               .collect(Collectors.toCollection(HashSet::new));
-          adjustedType = ResourceType.D2_URI;
         }
-        else
+
+        if (getWildcardResourceSubscribers().containsKey(originalType))
         {
-          adjustedType = originalType;
+          resources.add("*");
         }
-        _adsStream.sendDiscoveryRequest(adjustedType, resources);
-      }
 
-      for (ResourceType originalType: getWildcardResourceSubscribers().keySet())
-      {
-        ResourceType adjustedType = shouldSubscribeUriGlobCollection(originalType) ? ResourceType.D2_URI : originalType;
-        _adsStream.sendDiscoveryRequest(adjustedType, Collections.singletonList("*"));
+        if (resources.isEmpty())
+        {
+          continue;
+        }
+
+        Map<String, String> irv = _initialResourceVersionsEnabled ? getResourceVersions().get(adjustedType) : Collections.emptyMap();
```

Comment:
Nit: break up this long line
```suggestion
        Map<String, String> irv = _initialResourceVersionsEnabled
            ? getResourceVersions().get(adjustedType) : Collections.emptyMap();
```